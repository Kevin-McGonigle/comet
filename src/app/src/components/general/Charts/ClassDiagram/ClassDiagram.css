
#box {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 400px;
    height: 200px;
    color: white;
    font-family: 'Raleway', "Roboto", "Lucida Grande", Verdana, Arial, sans-serif;
    font-size: 12px;
}

.hrLine {
    width: 380px;
    margin-top: 4px;
    margin-bottom: 4px;
}

.classDiagramTable {
    --borderWidth: 3px;
    background: #11182f;
    position: relative;
    border-radius: var(--borderWidth);
}

.className {
    font-size: 20px;
    font-weight: bold;
    line-height: 1;
    text-align: center;
}

.classParent {
    text-align: center;
}

.classArgs {
}

.classFunctions {
}