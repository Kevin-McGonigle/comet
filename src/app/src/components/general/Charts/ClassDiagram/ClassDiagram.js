import React from 'react';
import styles from './ClassDiagram.css';
import classnames from 'classnames';
import {Pane, Popover} from 'evergreen-ui';

const cx = args => classnames(styles, args)

export const generateFunctionText = funcData => {
    return Object.keys(funcData).map(func => {
        const args = funcData[func].arguments;
        const ret = funcData[func].returnType;
        return `+ ${func}(${Object.keys(args).map(arg => `${arg} : ${args[arg]}`)}) : ${ret}\n`;
    })
}

export const generateClassArgsText = classArgs => {
    return Object.keys(classArgs).map(arg => <p>{`+ ${arg} : ${classArgs[arg]}`}</p>)
}

function ClassDiagram(node) {
    const {
        id,
        classArgs,
        classFunctions,
    } = node.data;

    return (
        <Popover
            content={
                <Pane
                    width={400}
                    height={200}
                    display="flex"
                    alignItems="center"
                    justifyContent="center"
                    flexDirection="column"
                >
                    <div className={cx('classDiagramTable')} id="box">
                        <div className={cx('className')}>{ id } - Cyclomatic Complexity 0</div>
                        <div className={cx('hrLine')}><hr/></div>
                        <div className={cx('classArgs')}>{ generateClassArgsText(classArgs) }</div>
                        <div className={cx('hrLine')}><hr/></div>
                        <div className={cx('classFunctions')}>{ generateFunctionText(classFunctions) }</div>
                    </div>
                </Pane>
            }>
            <Pane
                width={50}
                height={50}
                display="flex"
                alignItems="center"
                justifyContent="center"
                backgroundColor="#3498db"
            >
                {id}
            </Pane>
        </Popover>
    )
}

export default ClassDiagram;
