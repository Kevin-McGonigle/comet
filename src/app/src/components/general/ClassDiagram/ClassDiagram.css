
#box {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 400px;
    height: 200px;
    color: white;
    font-family: 'Raleway';
    font-size: 12px;
}

.hrLine {
    width: 380px;
    margin-top: 4px;
    margin-bottom: 4px;
}

.classDiagramTable {
    --borderWidth: 3px;
    background: #1D1F20;
    position: relative;
    border-radius: var(--borderWidth);
}

.className {
    font-size: 20px;
    font-weight: bold;
    line-height: 1;
    text-align: center;
}

.classParent {
    text-align: center;
}

.classArgs {
}

.classFunctions {
}