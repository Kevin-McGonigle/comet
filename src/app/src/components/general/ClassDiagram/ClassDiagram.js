import React from 'react';
import styles from './ClassDiagram.css';
import classnames from 'classnames';

const cx = args => classnames(styles, args)

export const generateFunctionText = funcData => {
    return Object.keys(funcData).map(func => {
        const args = funcData[func].arguments;
        const ret = funcData[func].returnType;
        return `+ ${func}(${Object.keys(args).map(arg => `${arg} : ${args[arg]}`)}) : ${ret}`;
    })
}

export const generateClassArgsText = classArgs => {
    return Object.keys(classArgs).map(arg => <p>{`+ ${arg} : ${classArgs[arg]}`}</p>)
}

const ClassDiagram = props => {
    const className = "ClassTest";
    const classParent = ['A', 'B', 'C'];
    const classArgs = {"Arg1": "String", "Arg2": "Int"}
    const classFunctions = {
        "Function1": {
            arguments: {
                "Arg1": "String",
                "Arg2": "Int",
            },
            returnType: "String"
    }};

    return (
        <div className={cx('classDiagramTable')} id="box">
            <div className={cx('className')}>{ className }</div>
            <div className={cx('hrLine')}><hr/></div>
            <div className={cx('classParent')}>Inherits from: { classParent }</div>
            <div className={cx('hrLine')}><hr/></div>
            <div className={cx('classArgs')}>{ generateClassArgsText(classArgs) }</div>
            <div className={cx('hrLine')}><hr/></div>
            <div className={cx('classFunctions')}>{ generateFunctionText(classFunctions) }</div>
        </div>
    )
};

export default ClassDiagram;
