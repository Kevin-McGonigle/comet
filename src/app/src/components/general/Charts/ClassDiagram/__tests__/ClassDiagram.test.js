import React from 'react';
import ClassDiagram, {generateClassArgsText, generateFunctionText} from '../ClassDiagram';
import {shallow} from 'enzyme';
import "../../../../../setupTests"

const nodeData = {
    data: {
        id: "A",
        classArgs: {"Arg1": "String", "Arg2": "Int"},
        classFunctions: {
            "Function1": {
                arguments: {
                    "Arg1": "String",
                    "Arg2": "Int",
                },
                returnType: "String"
            }
        }
    }
}

test('should render ClassDiagram as expected', () => {
    const component = shallow(
        ClassDiagram(nodeData)
    );
    expect(component.html()).toMatchSnapshot();
});

test('should generate Class Argument information as expected', () => {
    const generatedInfo = generateClassArgsText(nodeData.data.classArgs);
    expect(generatedInfo[0].props.children).toEqual("+ Arg1 : String");
    expect(generatedInfo[1].props.children).toEqual("+ Arg2 : Int");
})


test('should generate Class Function information as expected', () => {
    const generatedInfo = generateFunctionText(nodeData.data.classFunctions);
    expect(generatedInfo).toStrictEqual(['+ Function1(Arg1 : String,Arg2 : Int) : String']);
})