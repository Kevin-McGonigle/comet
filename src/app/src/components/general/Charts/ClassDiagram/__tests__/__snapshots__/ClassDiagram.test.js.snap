// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ClassDiagram as expected 1`] = `"<div role=\\"button\\" aria-haspopup=\\"true\\" class=\\"css-nil ðŸ“¦w_50px ðŸ“¦h_50px ðŸ“¦dspl_flex ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦bg-clr_3498db ðŸ“¦box-szg_border-box\\">A</div>"`;
